app/controllers/evaluations_controller.rb

```diff
@@ -28,6 +29,55 @@ def process_feedback
     end
   end
 
+  def export
+    if params[:provider].present?
+      mooc_provider = MoocProvider.find_by(name: params[:provider])
+      if mooc_provider.present?
+        courses = Course.where(mooc_provider: mooc_provider)
+      else # given provider not valid
+        courses = Course.all
+      end
+    else # no provider given
+      courses = Course.all
+    end
+    @courses_with_evaluations = []
+    courses.each do |course|
+      course_evaluations = Set.new
+      course.evaluations.each do |evaluation|
+        evaluation_object = {
+          rating: evaluation.rating,
+          description: evaluation.description,
+          creation_date: evaluation.created_at,
+          total_feedback_count: evaluation.total_feedback_count,
+          helpful_feedback_count: evaluation.positive_feedback_count,
+          course_status: evaluation.course_status.to_sym,
+          is_verified: evaluation.is_verified
+        }
+
+        if evaluation.rated_anonymously
+          evaluation_object[:user_name] = t('evaluations.anonymous')
```

Comment:
It's very good that you take care of anonymous evaluations. I just wondered, whether returning the String 'Anonymous' is good or not. On the one hand, it is how _we_ name these ratings, but on the other hand, the MOOC provider might want to change this to a different name. As long as we ensure that the name Anonymous won't change, we can go for it. Nevertheless, I would suggest to find a different solution without using I18n, as this might result in different words depending on the language (we use the HTTP Accept Parameter to select the language and this might result in an unintended behaviour here). For now, all our locals have the same translation, but as this is a subject of change, I would prefer another solution. Something I thought about is returning `null`, but this might not be the optimal solution. 

What do you think?